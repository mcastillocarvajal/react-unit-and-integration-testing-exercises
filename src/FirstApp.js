import React from 'react';
import PropTypes from 'prop-types';
// import React, { Fragment } from 'react';

const FirstApp = ( { greeting, subtitle } ) => {

    return (
        <>
            <h1> { greeting } </h1>
            {/* <h1> { JSON.stringify( greeting, null, 3 ) } </h1> */}
            <p>{ subtitle }</p>
        </>
    )
}

FirstApp.propTypes = {
    greeting: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    subtitle: 'Soy un subtitulo'
} 
export default FirstApp;