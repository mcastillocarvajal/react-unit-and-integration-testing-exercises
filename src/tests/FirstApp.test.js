import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from "enzyme";
import FirstApp from "../FirstApp";


describe('Prueba FirstApp', () => {
    
    // test('should show the msg Hola, soy Moises', () => {
    //     const saludo = 'Hola, soy Moises';
    //     const { getByText } = render( <FirstApp greeting={ saludo }/> );
    //     expect ( getByText( saludo ) ).toBeInTheDocument()
    // })

    test('should show <FirstApp /> correctly', () => {

        const saludo = 'Hola, soy Moises'

        const wrapper = shallow( <FirstApp greeting={ saludo } /> );

        expect( wrapper ).toMatchSnapshot();
    })
    
    test('should show the subtitle sent by props', () => {

        const saludo = 'Hola, soy Moises'
        const subtitle = 'Soy un subtitulo'

        const wrapper = shallow( 
            <FirstApp 
                greeting={ saludo }
                subtitle= { subtitle }
            /> 
        );

        const textoParrafo = wrapper.find('p').text();

        expect( textoParrafo ).toBe( subtitle )
        
    })
    
})
