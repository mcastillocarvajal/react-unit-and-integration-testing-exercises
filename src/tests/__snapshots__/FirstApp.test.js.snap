// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba FirstApp should show <FirstApp /> correctly 1`] = `
<Fragment>
  <h1>
     
    Hola, soy Moises
     
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
